src/LearnMe.Infrastructure/Repository/LessonsRepository.cs

```diff
@@ -0,0 +1,123 @@
+﻿using LearnMe.Infrastructure.Repository.Interfaces;
+using System.Threading.Tasks;
+using LearnMe.Infrastructure.Data;
+using System;
+using LearnMe.Infrastructure.Models.Domains.Lessons;
+using Microsoft.EntityFrameworkCore;
+
+namespace LearnMe.Infrastructure.Repository
+{
+    public class LessonsRepository : CrudRepository<Lesson>, ILessonsRepository
+    {
+        private readonly ApplicationDbContext _context;
+
+        private readonly ICalendarEventsRepository _calendarEventsRepository;
+
+        public LessonsRepository(
+            ApplicationDbContext context,
+            ICalendarEventsRepository calendarEventsRepository) : base(context)
+        {
+            _context = context;
+            _calendarEventsRepository = calendarEventsRepository;
+        }
+
+        public async Task<Lesson> CreateLessonAsync(string calendarId, Lesson lesson)
+        {
+            int? relatedEventDatabaseId = await FindRelatedCalendarEventDatabaseId(calendarId);
+
+            if (relatedEventDatabaseId != null)
+            {
+                lesson.CalendarEventId = (int)relatedEventDatabaseId;
+
+                return await InsertAsync(lesson);
+            }
+            else
+            {
+                return null;
+            }
+        }
+
+        public async Task<bool> DeleteLessonByCalendarIdAsync(string calendarId)
+        {
+            var toBeDeleted = await FindLessonByCalendarIdAsync(calendarId);
+
+            if (toBeDeleted != null)
+            {
+                _context.Remove(toBeDeleted);
+
+                return await SaveAsync();
+            }
+            else
+            {
+                return false;
+            }
+        }
+
+        public async Task<Lesson> GetLessonByCalendarIdAsync(string calendarId)
+        {
+            var found = await FindLessonByCalendarIdAsync(calendarId);
```

Comment:
FindLessons i FindLesson to w zasadzie takie same funkcje jak powyższe a tworzysz niepotrzebnie dwa taski na nie i je awaitujesz. Możesz to zrobić raz usuwająć te `Find` i ich implementacje wrzucić w tamte metody. No i dodaj `AsNoTracking()` zamiast `detached`. Właściwie po co to robisz?

Reply:
Mam 2 metody, bo `FindLessonByCalendarIdAsync(calendarId)` używam w kilku metodach - nie chciałam powtarzać kodu. Wg Ciebie lepiej kilka `await`-ów zamienić na powtarzanie kodu 3x ze względu na performance?

`detached` dodałam, bo wcześniej przeczytalam że tylko tak mogę nie trackować zmian przy pobieraniu 1 rekordu z bazy, ale chyba nie doczytałam. Zmienię na `AsNoTracking()`

Reply:
Później to opisze. Po prostu tutaj masz tak naprawde trzy awaity:

1. await w kontrolerze (metodzie)
2. await w metodzie GetLessons()
3. await w FIndLEssons. 

Teraz jest to zrobione źle bo to co siedzi w FindLessons powinno być w tej metogdzie `Get...` i nie będziesz powtarzać żadnego kodu. Zresztą w robocie się to czasami powtarza. Jak to poprawisz to zerkne jak to zrobiłaś i wtedy najwyżej podpowiem co dalej ale powinno być git. 

Ogólnie mnożenie asynców nie jest spoko jeśli tego nie musissz robić. W 90% jeśli chodzi o async/await nie ma to znaczenia jednak ta funckja "find" wygląda źle jeśli masz już metode "Get"

Reply:
Usunęłam `FindLessonByCalendarIdAsync(calendarId)` i metody, które wcześniej używały tego, teraz wyglądają tak:
![obraz](https://user-images.githubusercontent.com/19174792/100274592-fcd23d00-2f5e-11eb-9906-8e194ee2a102.png)

Reply:
+ jeszcze inna kwestia, tak bardziej koncepcyjnie. Pisałam Ci że mamy w encji CalenarEvent Id-z bazy i kolejną kolumnę CalendarId-który jest numerem z zewnętrznego serwisu (u nas Google). I powiem szczerze na razie wszędzie raczej korzystam z CalendarId, jego tez wystawiam w DTO, nie wystawiam naszego Id z bazy. Niestety już trochę w tym w kodzie zabrnęłam, ale tak się zastanawiam, czy tego nie zmienić i nie wystawić jeszcze Id z bazy w DTO? bo po Id z bazy mam indeksowanie (to jest primary key).... a po tej drugiej kolumnie nie. Co byś radził? dzięki

Reply:
Nie zwracamy na front tego, czego nie potrzebujemy. Dobrą praktyką jest mieć jakieś jedne, globalne DTO na encje i zawsze zwracać jedno DTO (ewentualnie uszczuplone coby front miał łatwiej, np nullowanie stringow)